### ISSUE TITLE
Constrain sampled parameters to user-defined subspaces

### ISSUE THREAD

**Issue by marvinschmitt** [bayesflow-org/bayesflow#209]
We should add a way for users to define that a certain parameter is bounded to a particular space and then enforce this constraint after sampling.

Current suggestions for an interface:

Pass a constraints object when sampling:

```python
def prior():
    theta = np.random.beta(a=1,b=1)
    return dict(theta=bf.constraints.Bounded(theta, lower=0, upper=1))
```

Pass constraints directly to the Approximator:

```python
approximator = bf.Approximator(
    ...,
    constraints=dict(theta="positive")
)
```

**Comment by paul-buerkner** [comment]
@LarsKue what is our current take on this? We had it in the new data_adapter, right?

**Comment by paul-buerkner** [comment]
So the current interface is in the data adapter using the pattern `data_adapter.constrain("theta", lower = 0, upper = 1)` for example. 

That said, we were thinking of allowing the constraints to optionally already definable in the `simulator`, which would add some meta-data to be later on read-out by the adapter.

**Comment by LarsKue** [comment]
This should have been auto-closed when #219 was merged. I will manually close the related issues.
